skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/projectgen/circuitgen/CircuitComponent.kt

```diff
@@ -0,0 +1,322 @@
+         fun create(screen: ggScreen, navigator: Navigator): FeaturePresenter {
+         }
+       }
+     }
+    */
+    val screenClassName = "${className}Screen"
+    val presenterClassName = "${className}Presenter"
+    val assistedInject = ClassName("dagger.assisted", "AssistedInject")
+    val assisted = ClassName("dagger.assisted", "Assisted")
+    val assistedFactory = ClassName("dagger.assisted", "AssistedFactory")
+
+    val factoryBuilder =
+      TypeSpec.interfaceBuilder("Factory").apply {
+        if (useAssistedInjection) {
+          addAnnotation(assistedFactory)
+        }
+      }
+
+    val constructorBuilder =
+      FunSpec.constructorBuilder().apply {
+        if (useAssistedInjection) {
+          addAnnotation(assistedInject)
+        }
+        addParameter(
+          ParameterSpec.builder("screen", ClassName("", screenClassName))
+            .apply {
+              if (useAssistedInjection) {
+                addAnnotation(assisted)
+              }
+            }
+            .build()
+        )
+        addParameter(
+          ParameterSpec.builder("navigator", ClassName("com.slack.circuit.runtime", "Navigator"))
+            .apply {
+              if (useAssistedInjection) {
+                addAnnotation(assisted)
+              }
+            }
+            .build()
+        )
+      }
+
+    val presenterClass =
+      TypeSpec.classBuilder(presenterClassName).apply {
+        if (noUi) {
+          addKdoc(
+            """
+         TODO (remove): This Circuit [Presenter] was generated without UI or [CircuitInject], and can be
+         used with legacy views via `by circuitState()` in your Fragment or Activity.
+          """
+              .trimIndent()
+          )
+        }
+        primaryConstructor(constructorBuilder.build())
+          .addSuperinterface(
+            ClassName("com.slack.circuit.runtime.presenter", "Presenter")
+              .parameterizedBy(ClassName(packageName, "${className}Screen.State"))
+          )
+          .addProperty(
+            PropertySpec.builder("screen", ClassName("", "${className}Screen"))
+              .addModifiers(KModifier.PRIVATE)
+              .initializer("screen")
+              .build()
+          )
+          .addProperty(
+            PropertySpec.builder("navigator", ClassName("com.slack.circuit.runtime", "Navigator"))
+              .addModifiers(KModifier.PRIVATE)
+              .initializer("navigator")
+              .build()
+          )
+          .addFunction(
+            FunSpec.builder("present")
+              .addModifiers(KModifier.OVERRIDE)
+              .addAnnotation(ClassName("androidx.compose.runtime", "Composable"))
+              .returns(ClassName("", "${className}Screen.State"))
+              .addCode(
+                """
+              val scope = rememberStableCoroutineScope()
+              return ${className}Screen.State() { event ->
+              }
+            """
+                  .trimIndent()
+              )
```

Comment:
Any reason we're generating this scope var? Instead I'm thinking we really just want to do this

```suggestion
              .addStatement("TODO(%S)", "Implement me!")
```